cmd/pr_status/pr_status.go

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2021 Skyscanner Limited.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package pr_status
+
+import (
+	"fmt"
+	"github.com/fatih/color"
+	"github.com/rodaine/table"
+	"github.com/skyscanner/turbolift/internal/campaign"
+	"github.com/skyscanner/turbolift/internal/github"
+	"github.com/skyscanner/turbolift/internal/logging"
+	"github.com/spf13/cobra"
+	"os"
+	"path"
+	"strings"
+)
+
+var reactionsOrder = []string{
+	"THUMBS_UP",
+	"THUMBS_DOWN",
+	"LAUGH",
+	"HOORAY",
+	"CONFUSED",
+	"HEART",
+	"ROCKET",
+	"EYES",
+}
+
+var reactionsMapping = map[string]string{
+	"THUMBS_UP":   "👍",
+	"THUMBS_DOWN": "👎",
+	"LAUGH":       "😆",
+	"HOORAY":      "🎉",
+	"CONFUSED":    "😕",
+	"HEART":       "❤️",
+	"ROCKET":      "🚀",
+	"EYES":        "👀",
+}
+
+var gh github.GitHub = github.NewRealGitHub()
+
+var list bool
+
+func NewPrStatusCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "pr-status",
+		Short: "Displays the status of PRs",
+		Run:   run,
+	}
+	cmd.Flags().BoolVar(&list, "list", false, "Displays a listing by PR")
+
+	return cmd
+}
+
+func run(c *cobra.Command, _ []string) {
+	logger := logging.NewLogger(c)
+
+	readCampaignActivity := logger.StartActivity("Reading campaign data")
+	dir, err := campaign.OpenCampaign()
+	if err != nil {
+		readCampaignActivity.EndWithFailure(err)
+		return
+	}
+	readCampaignActivity.EndWithSuccess()
+
+	skippedCount := 0
+	noPrCount := 0
+
+	openCount := 0
+	mergedCount := 0
+	closedCount := 0
+
+	reactions := make(map[string]int)
+
+	detailsTable := table.New("Repository", "State", "Reviews", "URL")
+	detailsTable.WithHeaderFormatter(color.New(color.Underline).SprintfFunc())
+	detailsTable.WithFirstColumnFormatter(color.New(color.FgCyan).SprintfFunc())
+	detailsTable.WithWriter(logger.Writer())
+
+	for _, repo := range dir.Repos {
```

Comment:
This is not for this PR, but since we're doing this loop in every single command more or less, perhaps we could set some wrapper around it with a callback function? WDYT? SHould i raise a ticket for this?

Reply:
Thinking about this again, would a wrapper function result in code that is _more_ concise and expressive than it is currently?

I wonder if there's a danger of replacing ~2 lines of explicit repeated code with a _different_ ~2 lines of repeated code that is less explicit.

Reply:
yeah i agree, in hindsight I think I liked the callback pattern so much I tried to fix all the problems with this new hammer.